src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -57,10 +66,50 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
       <ul>
     <%
       for (Message message : messages) {
+      /*%><div class = "profilenames" style= "color: green">
+        <%*/
         String author = UserStore.getInstance()
           .getUser(message.getAuthorId()).getName();
-    %>
-      <li><strong><%= author %>:</strong> <%= message.getContent() %></li>
+          %>
+      <li><div id="profileName"><%= author %>:</div>
+        <div id="commonInterests">
+          <%
+          
+          String result = "Common interests: ";
+          String currentUser = (String)request.getSession().getAttribute("user");
+
+          if (currentUser != null){
+            User currUser = UserStore.getInstance().getUser(currentUser);
```

Comment:
Completely optional design thought: It would also be possible to move more of this logic into the Servlet rather than having it in the JSP. The advantage would be that then it's in a "normal" Java file instead of mixed up with the HTML. Keeping the JSP a little simpler would make it clearer where "business logic" code that does heavier logical work is, as opposed to the HTML and CSS layout and display code. For bigger apps that large teams need to maintain, having that separation makes things easier to understand. https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller has some more information about this kind of design pattern.

The other advantage is that it would be easier to write tests. :stuck_out_tongue_winking_eye: 

Again, it's not necessary to make this change now! I just wanted to give you some extra background about design decisions like this. :+1: